core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java

```diff
@@ -148,6 +148,39 @@ public void testScriptsWithoutExtensions() throws IOException {
         }
     }
 
+    public void testHiddenFileSkipped() throws IOException {
+        ContextAndHeaderHolder contextAndHeaders = new ContextAndHeaderHolder();
+        buildScriptService(Settings.EMPTY);
+
+        logger.info("--> setup one hidden test file");
+        Path testFileHidden = scriptsFilePath.resolve(".hidden_file");
+        Path testRegularFile = scriptsFilePath.resolve("test_file.tst");
+        Streams.copy("test_hidden_file".getBytes("UTF-8"), Files.newOutputStream(testFileHidden));
+        Streams.copy("test_file".getBytes("UTF-8"), Files.newOutputStream(testRegularFile));
+        resourceWatcherService.notifyNow();
+
+        try {
+            logger.info("--> verify if hidden_file was skipped");
+            scriptService.compile(new Script("hidden_file", ScriptType.FILE, "test", null),
```

Comment:
This test is not really testing what we want to be testing here. The reason that it's not is because the cache key for a file named ".hidden_file" is not "hidden_file", but rather it is "". A file named ".hidden_file" never would have been processed by the compilation engine because it doesn't have an extension. So this will ultimately throw, but not for the right reason.